plugin/pkg/admission/security/podsecurity/admission_test.go

```diff
@@ -58,3 +71,123 @@ func TestConvert(t *testing.T) {
+	c := fake.NewSimpleClientset(
+		enforceImplicitPrivilegedNamespace,
+		enforcePrivilegedNamespace,
+		enforceBaselineNamespace,
+		enforceRestrictedNamespace,
+		warnBaselineNamespace,
+		warnRestrictedNamespace,
+		enforceWarnAuditBaseline,
+		warnBaselineAuditRestrictedNamespace,
+	)
+	p.SetExternalKubeClientSet(c)
+
+	informerFactory := informers.NewSharedInformerFactory(c, 0)
+	p.SetExternalKubeInformerFactory(informerFactory)
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	informerFactory.Start(stopCh)
+	informerFactory.WaitForCacheSync(stopCh)
+
+	if err := p.ValidateInitialization(); err != nil {
+		b.Fatal(err)
+	}
+
+	corePod := &core.Pod{}
+	v1Pod := &corev1.Pod{}
+	data, err := ioutil.ReadFile("testdata/pod.yaml")
+	if err != nil {
+		b.Fatal(err)
+	}
+	if err := yaml.Unmarshal(data, v1Pod); err != nil {
+		b.Fatal(err)
+	}
+	if err := v1.Convert_v1_Pod_To_core_Pod(v1Pod, corePod, nil); err != nil {
+		b.Fatal(err)
+	}
+
+	appsDeployment := &apps.Deployment{
+		ObjectMeta: metav1.ObjectMeta{Name: "mydeployment"},
+		Spec: apps.DeploymentSpec{
+			Template: core.PodTemplateSpec{
+				ObjectMeta: corePod.ObjectMeta,
+				Spec:       corePod.Spec,
+			},
+		},
+	}
+
+	namespaces := []string{
+		"enforce-implicit", "enforce-privileged", "enforce-baseline", "enforce-restricted",
+		"warn-baseline", "warn-restricted",
+		"enforce-warn-audit-baseline", "warn-baseline-audit-restricted",
+	}
+	for _, namespace := range namespaces {
+		b.Run(namespace+"_pod", func(b *testing.B) {
+			ctx := context.Background()
+			attrs := admission.NewAttributesRecord(
+				corePod.DeepCopy(), nil,
+				schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Pod"},
+				namespace, "mypod",
+				schema.GroupVersionResource{Group: "", Version: "v1", Resource: "pods"},
+				"",
+				admission.Create, &metav1.CreateOptions{}, false,
+				&user.DefaultInfo{Name: "myuser"},
+			)
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				if err := p.Validate(ctx, attrs, nil); err != nil {
+					b.Fatal(err)
+				}
+			}
+		})
+
+		b.Run(namespace+"_deployment", func(b *testing.B) {
+			ctx := context.Background()
+			attrs := admission.NewAttributesRecord(
+				appsDeployment.DeepCopy(), nil,
+				schema.GroupVersionKind{Group: "apps", Version: "v1", Kind: "Deployment"},
+				namespace, "mydeployment",
+				schema.GroupVersionResource{Group: "apps", Version: "v1", Resource: "deployments"},
+				"",
+				admission.Create, &metav1.CreateOptions{}, false,
+				&user.DefaultInfo{Name: "myuser"},
+			)
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				if err := p.Validate(ctx, attrs, nil); err != nil {
+					b.Fatal(err)
+				}
+			}
+		})
+	}
+	// deployment (small, normal, large)
+	// pod (small, normal, large)
+
+	// enforce privileged (implicit, no labels)
+	// enforce privileged (explicit)
+	// enforce baseline
+	// enforce+warn+audit baseline
+	// enforce restricted
+	// enforce+warn+audit restricted
+	// enforce baseline, warn+audit restricted
```

Comment:
Did you mean to leave these comments here? If so, add an explanation of what they mean.

Reply:
oops, removed